### DISCUSSION TITLE
How to view the order of the columns after reordering using DataTable reorderableColumns?

### DISCUSSION THREAD

**Discussion by Trelaquix** [primefaces/community#766] [PrimeReact]
Adding reorderableColumns to DataTable allows the order of the columns to be reordered by dragging and dropping. Is there anyway to log the order of the columns after the reordering?

**Comment by melloware** [comment]
Unfortunately it looks like its stored internall in the DataTable in a state value.

```js
const [columnOrderState, setColumnOrderState] = React.useState([]);
```

**Comment by Trelaquix** [comment]
I found a way to do it.

This was my initial DataTable:
`<DataTable
      reorderableColumns
      value={data}
      header={header}
      tableStyle={{ minWidth: "50rem" }}
    >
      {data}
    </DataTable>`

I added the onColReorder callback method to  DataTable and it works now.

Here is my new DataTable:
`<DataTable
      reorderableColumns
      value={data}
      header={header}
      tableStyle={{ minWidth: "50rem" }}
      onColReorder={onColReorder}
    >
      {data}
    </DataTable>`

My onColReorder method:
`const onColReorder = (event) => {
    // Handle column reorder event
    const reorderedColumns = event.columns.map((column) => column.props.header);
    console.log('Columns reordered:', reorderedColumns );
  };`

Now, logging reorderedColumns will show the new order of the columns.

**Reply by melloware** [reply]
Nice!
